### PR TITLE
neovide: 0b976c3 -> 4159c47f

### PR DESCRIPTION
**Pull request by ck3d**
Neovide moved from Vulkan to OpenGL (https://github.com/Kethku/neovide/pull/655) which has better GPU support.
This version should make https://github.com/NixOS/nixpkgs/issues/123262 obsolete.

The new version has one issues on my environment:

- The shift key only works for keys a-z. I can not type e.g. !@#?

If we merge this PR, Nix users can give feedback to the Neovide project.
But I am OK, when we wait for a more tested version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/neovim/neovide/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/neovide/default.nix:22]:
```suggestion
  version = "unstable-2021-06-18";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/neovim/neovide/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/neovide/default.nix:97]:
```suggestion
  postFixup = ''
      wrapProgram $out/bin/neovide \
        --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libglvnd xorg.libXcursor xorg.libXext xorg.libXrandr xorg.libXi ]}
```

File: pkgs/applications/editors/neovim/neovide/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/neovide/default.nix:64]:
```suggestion
  FONTCONFIG_FILE = makeFontsConf { fontDirectories = [ ]; };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by gebner** [comment]
> The new version has one issues on my environment:
> 
> * The shift key only works for keys a-z. I can not type e.g. !@#?

I've also experienced this issue the last time I tried out the opengl branch, this makes neovide pretty unusable (to be fair, the vulkan version doesn't run at all here so it would strictly speaking still be an improvement for me).

Maybe we should wait for this PR to be merged first?
https://github.com/Kethku/neovide/pull/736

**Comment by shaunsingh** [comment]
We will likely make a stable release after Kethku/neovide#736 is merged and linux keyboard issues are resolved. After that neovide should become much more usable on linux. I would personally wait until the stable release, but its up to you.

**Comment by ck3d** [comment]
Thanks for your comments. Lets wait for the keyboard issue.

**Comment by bnjmnt4n** [comment]
FYI: https://github.com/Kethku/neovide/pull/736 has been merged, and it works pretty well on my machine.

File: pkgs/applications/editors/neovim/neovide/default.nix
Comment (bnjmnt4n) [inline pkgs/applications/editors/neovim/neovide/default.nix:88]: I also added `wayland` and `libxkbcommon` to allow Neovide to run without XWayland on Sway.

**Comment by teto** [comment]
I had to change cargoSha256 and then I got
```
Validating consistency between /build/source//Cargo.lock and /build/neovide-unstable-2021-06-18-vendor.tar.gz/Cargo.lock
3,4d2
< version = 3
< 
1497a1496
>  "clap",

ERROR: cargoSha256 is out of date

Cargo.lock is not the same in neovide-unstable-2021-06-18-vendor.tar.gz

To fix the issue:
1. Use "0000000000000000000000000000000000000000000000000000" as the cargoSha256 value
2. Build the derivation and wait for it to fail with a hash mismatch
3. Copy the 'got: sha256:' value back into the cargoSha256 field
```

**Comment by tomcur** [comment]
With this patch applied to this PR, the latest Neovide commit (https://github.com/Kethku/neovide/commit/4159c47ff4f30073b92b9d63fc6ab70e07b74b6d) works on my machine.

```patch
--- a/default.nix	2021-06-22 13:22:11.259429723 +0100
+++ b/default.nix	2021-06-22 13:23:29.360111759 +0100
@@ -15,6 +15,7 @@
 , llvmPackages
 , makeFontsConf
 , libglvnd
+, libxkbcommon
 , xorg
 }:
 rustPlatform.buildRustPackage rec {
@@ -24,11 +25,11 @@
   src = fetchFromGitHub {
     owner = "Kethku";
     repo = "neovide";
-    rev = "599dc5887d8799ae8971259bc3bdbaa7b6e2ef45";
-    sha256 = "sha256-mwIJ9kI0N6W/3km8H6TdDfSO3TdLg+A/5DSxejRg3i8=";
+    rev = "4159c47ff4f30073b92b9d63fc6ab70e07b74b6d";
+    sha256 = "sha256-XwirJGXMGxc/NkpSeHBUc16ppvJ+H4ECnrOVu030Qfg=";
   };

-  cargoSha256 = "sha256-Fg2cDwjW6Ex5uUZ8226kgNd/6EoeJddp/rULqH70NRs=";
+  cargoSha256 = "sha256-w9xpNvdUZEOUeM+nTY7RjaVTYC9AahEMt5Igq37N7Bo=";

   SKIA_SOURCE_DIR =
     let
@@ -85,7 +86,7 @@

   postFixup = ''
       wrapProgram $out/bin/neovide \
-        --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libglvnd xorg.libXcursor xorg.libXext xorg.libXrandr xorg.libXi ]}
+        --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libglvnd libxkbcommon xorg.libXcursor xorg.libXext xorg.libXrandr xorg.libXi ]}
     '';

   postInstall = ''
```

**Comment by ck3d** [comment]
Thanks for the patch. The cargo hash was different, but the rest works. Regarding wayland support, please raise a separate PR. I can not test it.

**Comment by teto** [comment]
hell yeah that's freaking cool, neovide is finally usable here ! thanks for the work !

**Review by teto — APPROVED** [review]

**Comment by ck3d** [comment]
The outdated cargo hashes came from package name changes. A dependency which I didn't know.

**Comment by tomcur** [comment]
Thank you, it works!

**Review by tomcur — APPROVED** [review]

**Comment by ck3d** [comment]
Sorry, I did not recognize the draft state of my PR. Can somebody please merge this PR.

**Comment by bnjmnt4n** [comment]
:+1: I'll send a separate PR enabling Wayland support.

**Pull request merged**
